utils/python/CIME/XML/entry_id.py

```diff
@@ -56,6 +56,40 @@ def get_type_info(self, vid):
             return None
         else:
             return self._get_type_info(node)
+    
+    def get_default(self, node):
+        default = self.get_optional_node("default_value", root=node)
+        if default is not None:
+            return default.text
+        else:
+            return None
+        
+    
+    # Get type description , expect child with tag "type" for node         
+    def get_type (self, node):
+        type_node = self.get_optional_node("type", root=node)
+        if type_node is not None:
+            return type_node.text
+        else:
+            # Default to string
+            return "char"
+            
+    # Get description , expect child with tag "description" for parent node
+    def get_description (self, node):
+        type_node = self.get_optional_node("desc", root=node)
+        if type_node is not None:
+            return type_node.text
+        else:
+            return None
+
+    # Get group , expect child with tag "group" for parent node        
+    def get_group (self, node):
```

Comment:
I  suggest to change the name to _get_group(self,node) . It seems that methods starting with "_get" are taking a xml node as an argument where methods starting with "get" are taking an attribute name as argument and invoking the underscore method.